lib/psci/psci_private.h

```diff
@@ -38,6 +38,29 @@
 #include <psci.h>
 #include <spinlock.h>
 
+#if HW_ASSISTED_COHERENCY
+/*
+ * On systems with hardware-assisted coherency, make PSCI cache operations NOP,
+ * as PSCI participants are cache-coherent, and there's no need for explicit
+ * cache maintenance operations or barriers to coordinate their state.
+ */
+#define psci_flush_dcache_range(addr, size)
```

Comment:
#define psci_flush_dcache_range(...)

just in case the cache apis change?

OR we should just make it static inline function to ensure that the types are forced

static inline ... psci_flush_dcache_range (.. a, .. b)
{
#ifndef HW_ASSISTED_COHERENCY
flush_dcache_range(a, b);
#endif 
}

I am no good at github reviews.. so just trying to illustrate.